### PR TITLE
Remove decimal separators from values before calculating amounts when editing payments #5946

### LINKED ISSUE #5946: Editing item price of item on payment fails when item contains decimal separators
Make a payment and then go edit it, and set the value of one of the downloads to `1,000.99` and it'll look fine on initial save...but trying to edit it again will make the 'Download Total' fail and revert to $1. If you have the Dev Tools installed, the payment meta would have an invalid value for the `price` of the item as well.

### PR DESCRIPTION
**Pull request by cklosowski**
resolves #5946

### CODE REVIEW

File: includes/payments/class-edd-payment.php [hunk 1]
```diff
@@ -1405,14 +1405,17 @@ public function modify_cart_item( $cart_index = false, $args = array() ) {
...
-		$merged_item['price']    = $new_subtotal + $merged_item['tax'];
-
-		$this->cart_details[ $cart_index ]    = $merged_item;
-		$modified_download                    = $merged_item;
-		$modified_download['action']          = 'modify';
-		$modified_download['previous_data']   = $current_args;
+		// Format the item_price correctly now
+		$thousands_sep = edd_get_option( 'thousands_separator', ',' );
+
+		$merged_item['item_price'] = str_replace( $thousands_sep, '', $merged_item['item_price'] );
```
Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: Why not use `edd_sanitize_amount()` here?

Comment (cklosowski) [inline includes/payments/class-edd-payment.php]: yep, doing that now...forgot we had that already built ;)

File: includes/payments/class-edd-payment.php [hunk 2]
```diff
@@ -1405,14 +1405,17 @@ public function modify_cart_item( $cart_index = false, $args = array() ) {
...
-		$new_subtotal = $merged_item['item_price'] * $merged_item['quantity'];
-		$merged_item['subtotal'] = $new_subtotal;
-		$merged_item['price']    = $new_subtotal + $merged_item['tax'];
-
-		$this->cart_details[ $cart_index ]    = $merged_item;
-		$modified_download                    = $merged_item;
-		$modified_download['action']          = 'modify';
-		$modified_download['previous_data']   = $current_args;
+		// Format the item_price correctly now
+		$thousands_sep = edd_get_option( 'thousands_separator', ',' );
```
Comment (pippinsplugins) [inline includes/payments/class-edd-payment.php]: This line isn't needed now.

**Comment by cklosowski** [comment]
@pippinsplugins fixed.

**Pull request merged**